akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -751,25 +756,50 @@ private[akka] class ShardRegion(
   }
 
   def replyToRegionStatsQuery(ref: ActorRef): Unit = {
-    askAllShards[Shard.ShardStats](Shard.GetShardStats)
-      .map { shardStats =>
-        ShardRegionStats(shardStats.map {
-          case (shardId, stats) => (shardId, stats.entityCount)
-        }.toMap)
+    queryShards[ShardStats](shards, Shard.GetShardStats)
+      .map { qr =>
+        // TODO add ShardRegionStats field for timeouts (proto): ShardRegion.ShardRegionStats(stats, timeouts.tosSet)
+        // Until then
+        ShardRegionStats(qr.responses.map(stats => (stats.shardId, stats.entityCount)).toMap ++ qr.timedout.map((_, 0)))
       }
       .recover {
-        case _: AskTimeoutException => ShardRegionStats(Map.empty)
+        case _: AskTimeoutException =>
+          log.warning("{} shards queried but timed out after {}.", shards.size, settings.shardRegionQueryTimeout)
```

Comment:
Can this still happen? As the timeouts are already recovered inside of `askOne`?

Reply:
No, do you suggest removing?

Reply:
If it can't happen anymore it'd be better to remove it I think

Reply:
It does not log below in `queryShards`, I just didn't think I should change the original `recover` but it may make sense.